[Au Jun Ming Jeremiah] iP

Jolonauh opened the pull request:
# MicroManager
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MicroManager frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/Jolonauh/ip).
2. double-click it. 
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Rename chatbot

Commit:
Add CLI UserInterface interface

Commit:
Implement echo functionality

Commit:
Rename getUserInput method in ConsoleUserInterface to handleUserInput

Commit:
Implement functionality to add and list tasks

Commit:
Add ability to mark and unmark tasks as done

Commit:
Add task types: todos, events and deadlines

Commit:
Add automated text ui testing

Commit:
Add error handling

Commit:
Update test cases

Commit:
Add delete task functionality

Commit:
remove total field and related methods in Task class

Commit:
Rename UI class and add additional classes

Commit:
Add new classes

Commit:
Add dates and times

Commit:
Add find command

Commit:
Merge branch 'branch-Level-9' into add-gradle-support

Commit:
Add gradle support

Commit:
Add find command

Commit:
Add tasks.txt

Commit:
Merge branch 'master' of https://github.com/Jolonauh/ip

# Conflicts: tasks.txt updated
#	data/tasks.txt

radeon2525 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,50 @@
+        this.filepath = filepath;
+    }
+
+    public List<Task> load() throws DukeException {
```
Maybe a more specific name like loadToFile would be better suited

radeon2525 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,50 @@
+        return taskList;
+    }
+
+    public void update(TaskList taskList) throws DukeException {
```
A better name can be 'updateToFile'

yespiggy replied:
Agree. Perhaps a more specific method name is better?

radeon2525 commented on src/main/java/duke/dataprocessing/CommandParser.java:
```diff
@@ -0,0 +1,94 @@
+
+import java.time.LocalDate;
+
+public class CommandParser {
```
I like how you have done implementation for switch case

radeon2525 reviewed (commented):
Overall, well structured project with not much coding standard violations. Maybe can just try to include some JavaDocs for better readability

yespiggy commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,35 @@
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        TaskList searchResults = new TaskList();
+        for (int i = 0; i < taskList.size(); i++) {
+            Task currTask = taskList.get(i);
```
Perhaps a more specific name rather than `currTask` might be better?

yespiggy commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,14 @@
+import duke.storage.Storage;
+import duke.storage.TaskList;
+import duke.ui.Ui;
+
```
I like the way you handle different types of commands. 👍 👍

yespiggy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Command c = CommandParser.parse(fullCommand);
```
It might be better to have a more detailed name, instead of `c`?

yespiggy reviewed (commented):
Great job overall! There are just some minor issues with some names not being descriptive enough.

Commit:
Add JUnit tests

Commit:
Make jar file

Commit:
Change code to fit coding standard

Commit:
Add GUI

Commit:
Add assertions to necessary classes

Assertions were added to some classes to ensure that certain
conditions were fulfilled for some methods during execution.

Commit:
Improve code quality in project

Aspects of code quality, such as long methods, complicated expressions
and poor SLAP were identified and rectified in code.

Jolonauh committed:
Merge pull request #2 from Jolonauh/branch-A-Assertions

Add assertions

Commit:
Add update extension

Commit:
Add Ui.png

Commit:
Add user guide
